monai/data/utils.py

```diff
@@ -240,7 +241,57 @@ def list_data_collate(batch: Sequence):
     """
     elem = batch[0]
     data = [i for k in batch for i in k] if isinstance(elem, list) else batch
-    return default_collate(data)
+    try:
+        return default_collate(data)
+    except RuntimeError as re:
+        re_str = str(re)
+        if "stack expects each tensor to be equal size" in re_str:
+            re_str += (
+                "\nMONAI hint: if your transforms intentionally create images of different shapes, creating your "
+                + "`DataLoader` with `collate_fn=pad_list_data_collate` might solve this problem (check its "
+                + "documentation)."
+            )
+        raise RuntimeError(re_str)
+
+
+def pad_list_data_collate(batch: Sequence):
```

Comment:
this mainly support `batch` with a list of dictionary, shall we support list of list as well? the underlying `default_collate` is quite powerful: https://github.com/pytorch/pytorch/blob/master/torch/utils/data/_utils/collate.py#L43

Reply:
Although this is doable, I can't immediately see the use case. Currently I only want to decollate for inverse transformations which means the format is list of dictionaries. Would you accept this?

```python
if isinstance(batch[0], list):
    raise NotImplementedError("decollation has only currently been implemented for list of dictionaries. Please raise a PR if you feel this functionality would be useful")
```

Reply:
sure, sounds great

Reply:
In the end it was pretty simple to allow for list of lists/tuples, so I added that too.

Reply:
thanks, forgot to say that we should support different padding mode here...

Reply:
Good point, I've just added padding options